modules/compute/gke-node-pool/variables.tf

```diff
@@ -172,11 +172,36 @@ variable "spot" {
 }
 
 variable "compact_placement" {
-  description = "Places node pool's nodes in a closer physical proximity in order to reduce network latency between nodes."
+  description = <<-EOT
+  Places node pool's nodes in a closer physical proximity in order to reduce network latency between nodes.
+  Use `placement_type` and `placement_policy` if you want to use the placement policy you created.
+  Note that `compact_placement` might be deprecated in future in favour of `placement_type` and `placement_policy`
+  EOT
   type        = bool
   default     = false
 }
 
+variable "placement_type" {
+  description = <<-EOT
+  Type of the group placement to use for the node pool's nodes. This is used together with `placement_policy`.
+  `COMPACT` is the only supported value currently.
+  Note that `placement_type` and `placement_policy` take effect only when `compact_placement` is `false`.
+  EOT
+  type        = string
+  default     = null
+
+}
+
+variable "placement_policy" {
+  description = <<-EOT
+  Name of the placement policy to use when `placement_type` is set.
+  It is assumed that the specified policy exists. To create a placement policy refer to https://cloud.google.com/sdk/gcloud/reference/compute/resource-policies/create/group-placement.
```

Comment:
Perhaps as a follow up item, we should also test whether this policy exists or not (if that is possible by getting this resource)

Reply:
Not validating it separately as the node pool provision operation is supposed to fail for an unknown policy.

But, rethinking about it. Validating and failing early would be a good experience than failing at the time of provisioning the node pool (which happens after cluster creation). Will address as a follow-up 👍